src/stamps.ts

```diff
@@ -236,65 +260,88 @@ export class StampsManager {
       const stamps = await beeDebug.getAllPostageBatch()
       logger.debug('retrieved stamps', stamps)
 
-      const { amount, ttlMin, depth } = config
+      const { depth, amount, ttlMin, usageThreshold } = config
 
       // Get all usable stamps sorted by usage from most used to least
-      this.usableStamps = filterUsableStampsExtends(stamps)
+      this.usableStampsExtendsTTL = filterUsableStampsExtendsTTL(stamps)
 
       if (!this.isBuyingStamp) {
-        if (this.usableStamps.length === 0) {
+        const minTimeThreshold = ttlMin + config.refreshPeriod / 1000
+
+        if (!isNaN(depth) && this.usableStampsExtendsTTL.length === 0) {
           this.isBuyingStamp = true
           try {
             const { stamp: newStamp } = await buyNewStamp(depth, amount, beeDebug)
 
             // Add the bought postage stamp
-            this.usableStamps.push(newStamp)
+            this.usableStampsExtendsTTL.push(newStamp)
           } finally {
             this.isBuyingStamp = false
           }
         } else {
-          await this.verifyUsableStamps(beeDebug, ttlMin, config, amount)
+          await this.verifyUsableStamps(beeDebug, minTimeThreshold, amount)
+        }
+      }
+
+      this.usableStampsExtendsCapacity = filterUsableStampsExtendsCapacity(stamps, usageThreshold)
+
+      for (const stamp of this.usableStampsExtendsCapacity) {
+        try {
+          if (!this.topingUpStamps.includes(stamp.batchID)) {
+            logger.info(`extending stamp capacity: ${stamp.batchID}`)
+
+            this.topingUpStamps.push(stamp.batchID)
+            const stampRes = await topUpStamp(beeDebug, stamp.batchID, BigInt(stamp.amount).toString())
+            setTimeout(() => this.completeTopUp('capacity', stampRes), 60000)
+            await beeDebug.diluteBatch(stamp.batchID, stamp.depth + 1)
+            logger.info(`capacity extended for stamp ${stamp.batchID}`)
+          }
+        } catch (err) {
+          logger.error('failed to extend stamp capacity', err)
         }
       }
     } catch (e) {
       logger.error('failed to refresh on extends postage stamps', e)
     }
   }
 
-  async verifyUsableStamps(
-    beeDebug: BeeDebug,
-    ttlMin: number,
-    config: StampsConfigAutobuy | StampsConfigExtends,
-    amount: string,
-  ) {
-    for (let i = 0; i < this.usableStamps!.length; i++) {
```

Comment:
The fact that you have replaced the `this.usableStamps` for `this.usableStampsExtendsTTL` makes me think that you have broken the `autobuy` mode with it. Isn't it true?

Reply:
No, this was resolved by using a separate one for autobuy. But it was completely modified by the refactor